### PR TITLE
fix combine=nil and special key output to every shard

### PR DESCRIPTION
**Pull request by YuchaoZheng**

### CODE REVIEW

File: executor/executor.go [hunk 1]
```diff
@@ -27,6 +27,14 @@ type ComputeWrapClass struct {
 	combiner mapred.Combiner
 }
 
+var WhetherKeyOutPutToEvertShard func(keyBytes []byte) bool
```
Comment (Lcch) [inline executor/executor.go]: It should be wrapped into ComputeWrapClass just like mapper, reducer, combiner. It shouldn't be a global variable.

Comment (Lcch) [reply executor/executor.go]: Also add comment about what it is for please.

File: executor/executor.go [hunk 2]
```diff
@@ -164,20 +172,37 @@ func (cw *ComputeWrapClass) DoMap(rr records.RecordReader, writers []records.Rec
...
-				writers[rBucketID].WriteRecord(curRecord)
+				if HasOutputToEveryShard == false || WhetherKeyOutPutToEvertShard == nil || WhetherKeyOutPutToEvertShard(curRecord.Key) == false {
+					HasOutputToEveryShard = false
+					rBucketID := util.HashBytesKey(curRecord.Key) % nReduce
+					writers[rBucketID].WriteRecord(curRecord)
+				} else {
+					if WhetherKeyOutPutToEvertShard(r.Key) {
+						r.Value = CombineValueOutPutToEvertShard(curRecord.Value, r.Value)
+					} else {
+						HasOutputToEveryShard = false
```
Comment (Lcch) [inline executor/executor.go]: It is possible that there are multiple keys which satisfy "WhetherKeyOutPutToEvertShard", in this case, it will have bug.

Comment (YuchaoZheng) [reply executor/executor.go]: 当时我想的是只有一个特殊的key,刚才跟德康老师交流了下，会有很多特殊的key，我修改下

**Comment by Lcch** [comment]
Closed, reopen on #59

**Comment by Lcch** [comment]
Closed, reopen on #59

**Pull request closed without merging**